src/libs/actions/IOU.js

```diff
@@ -887,8 +888,8 @@ function requestMoney(
         onyxData,
     );
     resetMoneyRequestInfo();
-    Navigation.dismissModal(isMoneyRequestReport ? report.reportID : chatReport.reportID);
-    Report.notifyNewAction(chatReport.reportID, payeeAccountID);
+    Navigation.dismissModal(activeReportID);
+    Report.notifyNewAction(activeReportID, payeeAccountID);
```

Comment:
why do we change that from chatReport.reportID to activeReportID? Can you please explain

Reply:
@narefyev91 We are always subscribed to the report opened in the report screen here:
https://github.com/Expensify/App/blob/70813dbf3e2f73a73e52f880312bc198b39f4dec/src/pages/home/report/ReportActionsList.js#L257

So when we are in an expense report, `chatReport.reportID` which corresponds to `currentChatReport` passed to `getMoneyRequestInformation`
```js
const currentChatReport = isMoneyRequestReport ? ReportUtils.getReport(report.chatReportID) : report;
```

So, `currentChatReport` actually becomes the original report with the other person and not the IOU or expense report. This means that when we notify a new action, we do it on parent report and not the actual expense report when using `chatReport.reportID`.
This is why we modified the `Navigation.dismissModal` line to use a different `reportID` so that the user is redirected to the expense report from where they requested the money. The same logic should be used for `notifyNewAction`.